doc/design/vic-machine.md

```diff
@@ -0,0 +1,180 @@
+
+1. input manifest (if omitted, this creates a new manifest from scratch)
+2. a restriction set of some kind ( e.g. compute resource path, datastore prefix, registry whitelist)
+  - as a note, user credentials are also considered a restriction as they control permissible operations
+
+
+The _viadmin_ uses vic-machine to create a _base manifest_ containing the following restrictions:
+* target vSphere environment
+* vSphere user and credential for created VCHs:
+  - must already exist, or
+  - must be created by vic-machine during manifest creation, or
+  - demand created during VCH deployment, requiring stored admin credentials:
+* stored _viadmin_ credentials if necessary:
+    - encrypted credentials and [validating proxy](components.md#validating-proxy) URI, or
+    - unencrypted credentials (_viadmin_ and _admin_ roles are held by the same entity and manifests are stored securely)
+
+The _base manifest_ is the minimum set of informtion necessary from the _viadmin_ role. There is additional information that will almost always be required for full function of a VCH:
+* vSphere network to use for container network
+  - must already exist
+* vSphere switch to use for container network:
+  - must already exist
+  - if switch, port group or network is demand created during VCH deployment, requiring stored admin credentials
+    - in an enterprise environment, creation of a portgroup without VXLAN or VLAN is unlikely to suffice, so pre-existing portgroup is highly recommended
+
+To create a VCH, disambiguation of which resource to use from the available set is necessary. If there is no ambiguity, i.e only one of each, then this can be omitted:
+* _client_ network - necessary for any access to the VCH
+* default container network:
+  - _private_ port group or dedicated vSwitch, or
+  - IPAM config for _client_ network (defaults to DHCP)
+* datastore paths for (may all be the same path):
+  - images
+  - containers
+  - volumes
+
+
+## Implementation approach
+
+My current approach is to have several components that get rolled into both vic-machine. Initial implementation should focus on:
+* CLI parsing into config structs
+* validation of config struct values
+* reification of VCH from config struct
+
+Components:
+* command line argument parsing
+  - maps command line arguments to internal config data structures
+* validation of config structure values
+  - given an internal config data structure it's necessary to check the specifics against the target vSphere to ensure they are valid - past validity is no indicator of current validity
+  - this may also translate from symbolic names to morefs, resulting in a manifest that is resilient to name changes, but fragile across vSphere instances with identical naming schemes. This should be a user selectable behaviour, I prefer it on by default.
```

Comment:
s/behaviour/behavior/ 

"I prefer it on by default" -- is this George speaking or the project? I'd recommend rewording that phrase or removing it and leaving "This should be user selectable behavior."